feat: contract testing: add missing output checks (#650)
stoplightio/prism: docs/guides/client.md

# Prism Client

> Prism inclues an Axios style HTTP Client that you can use to seamlessy perform requests to both a real server and a mocked document.

Reviewer: ```suggestion
Prism includes an full featured HTTP Client that you can use to seamlessly perform requests to both a real server and a mocked document. The client is modeled after Axios so it may feel familiar.
```

Author: I thought it was a good idea to say we inspired to Axios?

Author: Ignore me, your suggestion includes that.

> Prism includes an full featured HTTP Client that you can use to seamlessly perform requests to both a real server and a mocked document. The client is modeled after Axios so it may feel familiar.

Reviewer 2: `a full
```suggestion
Prism includes a fully-featured HTTP Client that you can use to seamlessly perform requests to both a real server and a mocked document. The client is modeled after Axios so it may feel familiar.
```

> ### Create from filename or http resource

Reviewer 2: ```suggestion
### Create from a filename or http resource
```

```ts
const client = await createClientFromResource('examples/petstore.oas2.yaml', {
  mock: true,
  validateRequest: true,
  validateResponse: true,
});
```

> ### Create from OAS document in string format

Reviewer: ```suggestion
### Create from OpenAPI string
```

Remember that we don't really talk about OAS for SEO reasons, and human readability. OpenAPI is a known brand, so whilst you might use oas for internal variables we generally want to say OpenAPI on anything user facing.

```ts

> const oasFile = `

Reviewer: ```suggestion
const descriptionDoc = `
```

openapi: 3.0.2
paths:
  /hello:
    get:
      responses:
        200:
          description: hello
`;

> const client = await createClientFromString(oasFile, {

Author: ```suggestion
const client = await createClientFromString(descriptionDoc, {
```

  mock: true,
  validateRequest: true,
  validateResponse: true,
});
```

> ### Create from manual Http Operations (you might want _NOT_ use this)

Reviewer: Keep titles "Short, Sweet, And Using Title Case"

> ### Create from manual Http Operations (you might want _NOT_ use this)

Author: ```suggestion
### Create From Manual Http Operations
```

```ts
const client = createClientFromOperations(
  [
    {
      method: 'get',
      path: '/hello',
      id: 'n1',
      responses: [{ code: '200' }],
    },
  ],
  { mock: true, validateRequest: true, validateResponse: true }
);
```

---

Once you've got a client instance:

1. You can perform the request using the generic method:

```ts
client.request('https://google.it', { method: 'get' }).then(response => console.log(response));
```

> The response object has all the informations you need, including the used configuration object.

Reviewer 2: ```suggestion
The response object has all the information you need, including the used configuration object.
```

2. You can override the configuration object on the request level if you prefer

```ts
client
  .request('https://google.it', { method: 'get' }, { validateResponse: false })
  .then(response => console.log(response));
```

> This disables the validation response _only for the current request_

Reviewer 2: ```suggestion
This disables response validation _only for the current request_
```
